test/Main.hs

```diff
@@ -536,6 +537,19 @@ testDouble TestEnv{..} = do
     [Only (x :: Double)] <- query_ conn "SELECT '-Infinity'::float8"
     x @?= (-1 / 0)
 
+testParenNegatives :: TestEnv -> Assertion
+testParenNegatives TestEnv{..} = do
+    [Only (x :: Int)] <- query conn "SELECT ?::int2" (Only (-32768 :: Int))
+    x @?= -32768
+    [Only (x :: Int)] <- query conn "SELECT ?::int2" (Only (-32768.4 :: Double))
```

Comment:
why not `SELECT (?) :: int2`?

I'm not yet convinced this **must** be in the library

Reply:
That works for this one. It wouldn't work for this:

```haskell
query conn "SELECT * FROM ? tbl"
  (Only $ Values ["int2", "int2"]
                 [(-32768 :: Integer, -32768.4 :: Float)]
  )
```

In general, I expect users of this library will be using `ToField` instances as though they produce atoms. The documentation doesn't quite say so explicitly, but that's the strong vibe I get; e.g. it doesn't typically write `(?)`, which I'd expect if "`toField` might not produce an atom" was standard. The `ToField (Values a)` instance assumes that, too.